### PR TITLE
Update for latest lenthall

### PR DESCRIPTION
**Pull request by geoffjentry**

### CODE REVIEW

**Comment by cjllanwarne** [comment]
Won't compile:

[error] /Users/chrisl/IdeaProjects/cromwell/src/main/scala/cromwell/instrumentation/Instrumentation.scala:20: value getConfigOption is not a member of com.typesafe.config.Config
[error]   private val Config = ConfigFactory.load.getConfigOption("instrumentation")
[error]                                           ^
[error] one error found
[error] (compile:compileIncremental) Compilation failed

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala [hunk 1]
```diff
@@ -197,19 +197,15 @@ object WorkflowDescriptor {
...
-  def lookupDockerHash(conf: Config) = lookupWithDefault(conf, "call-caching", "lookup-docker-hash", DefaultLookupDockerHash)
-  // TODO: Add to lenthall
-  // FIXME: There's an identicaly named function in ConfigUtil which doesn't behave quite the same
-  def getConfigOption(conf: Config, key: String): Option[Config] = if (conf.hasPath(key)) Option(conf.getConfig(key)) else None
+  private def configCallCaching(conf: Config) = lookupBooleanWithDefault(conf, "call-caching", "enabled", DefaultCallCachingValue)
+  private def lookupDockerHash(conf: Config) = lookupBooleanWithDefault(conf, "call-caching", "lookup-docker-hash", DefaultLookupDockerHash)
 
-  // FIXME/TODO: Should live in ConfigUtil but getConfigOption is weird, see above
-  def lookupWithDefault(conf: Config, stanza: String, key: String, default: Boolean) = {
+  private def lookupBooleanWithDefault(conf: Config, stanza: String, key: String, default: Boolean) = {
```
Comment (cjllanwarne) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: The TODO is TOGONE but this is still not in Lenthall... Also is there a reason this function isn't a nice generic `lookupWithDefault[A](... default: A)`?

**Comment by cjllanwarne** [comment]
Aside from it not compiling, this looks good :+1:

File: src/main/scala/cromwell/engine/WorkflowDescriptor.scala (hunk 1 above)
Comment (geoffjentry) [inline src/main/scala/cromwell/engine/WorkflowDescriptor.scala]: Hah yes. This is actually the reason I started updating lenthall :)  Then it turned out that they already had a similar concept of get w/ default but it wasn't quite the same (wouldn't reach into an arbitrary stanza) and I got too lazy to come up with a decent name to distinguish the difference. Then I decided this is such a tiny amount of sugar that it wasn't worth the hassle anyways

**Comment by Horneth** [comment]
:+1:

**Pull request merged**